lib/main_common.pm

```diff
@@ -2519,6 +2585,88 @@ sub set_mu_virt_vars {
     bmwqemu::save_vars();
 }
 
+sub set_sles16_mu_virt_vars {
+    # SLES16 MU testing only supports staging mode
+    # - FLAVOR contains "staging" AND INCIDENT_REPO is set → Staging mode
+
+    if (get_var('FLAVOR', '') =~ /staging/i && get_var('INCIDENT_REPO')) {
+        # Staging mode: Use dedicated virt staging profile for KVM patterns
+        set_var('INST_AUTO', 'virtualization/agama_virt_auto/sle_virt_default_staging.jsonnet');
+        diag("Staging mode detected (FLAVOR contains staging and INCIDENT_REPO is set): using virtualization/agama_virt_auto/sle_virt_default_staging.jsonnet for installation");
+    } else {
+        # No valid mode detected, exit test immediately
+        die("SLES16 MU test requires staging mode - FLAVOR must contain 'staging' with INCIDENT_REPO set");
+    }
+
+    # Parse BUILD variable and set UPDATE_PACKAGE (format example: BUILD=:345:qemu)
+    # This logic is the same as set_mu_virt_vars() to maintain consistency
+    if (!get_var('UPDATE_PACKAGE') && (my $BUILD = get_var('BUILD'))) {
+        $BUILD =~ /^:(\d+):([^:]+)$/im;
+        die "BUILD value is $BUILD, but does not match required format." unless $2;
+
+        set_var('UPDATE_PACKAGE', $2);
+        diag("BUILD is $BUILD, UPDATE_PACKAGE is set to " . get_var('UPDATE_PACKAGE', ''));
+    }
+
+    # Parse and set the target package for host installation verification
+    # Package name transformation logic (same as SLES12/15 MU testing)
+    my $_pkg = get_var('UPDATE_PACKAGE', '');
+    my $_update_package = '';
+
+    if ($_pkg) {
+        if ($_pkg =~ /none/) {
+            # 'none' is for ease of functional testing when no package update is needed
+            $_update_package = '';
+        } elsif ($_pkg =~ /qemu|virt-manager|libguestfs|libslirp|open-vm-tools|snphost/) {
+            # Direct package name usage (SLES16 currently supports KVM/QEMU only)
+            $_update_package = $_pkg;
+        } elsif ($_pkg =~ /snpguest/) {
+            # SNP guest testing requires patching workflow
+            $_update_package = $_pkg;
+            set_var('PATCH_ON_GUEST', 1);
+        } elsif ($_pkg =~ /poppler/) {
```

Comment:
Would you give more information how poppler will be needed in our triggered VT test? In qem-bot/metadata, do we need to enable test for it?

Reply:
this is test package from core team, it is convenient for testing.
I will remove it before merging.